arduino-core/src/cc/arduino/packages/uploaders/SerialUploader.java

```diff
@@ -280,12 +280,18 @@ private String waitForUploadPort(String uploadPort, List<String> before, boolean
   private boolean uploadUsingProgrammer(String buildPath, String className) throws Exception {
 
     TargetPlatform targetPlatform = BaseNoGui.getTargetPlatform();
+    if (targetPlatform == null) {
+      throw new RunnerException(tr("Gould not load platform for programmer. Ensure programmer selection is correct for the board."), false);
+    }
     String programmer = PreferencesData.get("programmer");
     if (programmer.contains(":")) {
       String[] split = programmer.split(":", 2);
       targetPlatform = BaseNoGui.getCurrentTargetPlatformFromPackage(split[0]);
```

Comment:
I think this could again return null when an invalid package is specified in the "programmer". Or does this throw an exception instead?

Reply:
As in the other comment. Trying to be defensive AND give a unique error message for future users who ask for support.

Reply:
I'm a bit confused. I think this can return `null` *without* that resulting in a proper error message (but still a NullPointerException). Maybe you misunderstood my meaning?